### PR TITLE
command: add `-connect` flag to job init

### LINKED ISSUE #6127: Add `-connect` and `-connect -short` flags to `nomad job init`
Nomad should provide an example job similar to the one provided in the preview guide to demonstrate Connect’s features. Add a `-connect` flag to `nomad job init` command to generate a connect version of the existing init jobs.

```
nomad init -connect
```

This will compose with the `-short` flag which omits comments:

```
nomad init -connect -short
```

### PR DESCRIPTION
**Pull request by tgross**
Fixes #6127

Adds an example job for Consul Connect integration as well as an annotated example job. Some of the text was taken out of the RFC and the preview documentation, so we might want to adjust that as we refine that documentation.

### CODE REVIEW

File: command/assets/connect.nomad [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+        # required and a dynamic port will be registered for remote services
+        # to connect to Envoy with the name `connect-proxy-<service>`.
+        #
+        # By default, Envoy will be run via its official upstream Docker image.
+        # The optional "task" field overrides the default Envoy task with a
+        # user-defined task with the `kind = "connect-proxy:<service>"` field
+        # set.
+        #
+        sidecar_service {
+          # task = "connect-count-api"
```
Comment (nickethier) [inline command/assets/connect.nomad]: Whats the purpose of this comment?

File: command/assets/connect.nomad [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+
+      connect {
+        # The "sidecar_service" stanza configures the Envoy sidecar admission
+        # controller. For each task group with a sidecar_service, Nomad  will
+        # inject an Envoy task into the task group. A group network will be
+        # required and a dynamic port will be registered for remote services
+        # to connect to Envoy with the name `connect-proxy-<service>`.
+        #
+        # By default, Envoy will be run via its official upstream Docker image.
+        # The optional "task" field overrides the default Envoy task with a
```
Comment (nickethier) [inline command/assets/connect.nomad]: This is no longer the case. We're working on a different method of overriding the Envoy task

File: command/assets/connect.nomad [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+    # The Envoy sidecar admission controller will inject an Envoy task into
+    # any task group for each service with a sidecar_service stanza it contains.
+    # A group network will be required and a dynamic port will be registered for
+    # remote services to connect to Envoy with the name `connect-proxy-<service>`.
+    # By default, Envoy will be run via its official upstream Docker image.
+    #
+    # There are two ways to modify the default behavior:
+    #   * Tasks can define a `sidecar_task` stanza in the `connect` stanza
+    #     that merges into the default sidecar configuration.
+    #   * Tasks can use the `sidecar_service` stanza to refer to a task
```
Comment (nickethier) [inline command/assets/connect.nomad]: You don't need to refer to it in the sidecar_service stanza. You just need to make sure the kind is set correctly.

File: command/assets/connect.nomad [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+        #     memory = 512
+        #   }
+        # }
+      }
+    }
+
+    task "dashboard" {
+      driver = "docker"
+
+      # the application can take advantage of automatically created
```
Comment (nickethier) [inline command/assets/connect.nomad]:
```suggestion
      # The application can take advantage of automatically created
```

File: command/assets/connect.nomad (hunk 2 above)
Comment (tgross) [reply command/assets/connect.nomad]: Is that https://github.com/hashicorp/nomad/issues/6040? Maybe we should wait on wrapping up this PR until that work has been solidified?

File: command/assets/connect.nomad (hunk 1 above)
Comment (tgross) [reply command/assets/connect.nomad]: As example of `# The optional "task" field overrides the default Envoy task with a...` above. But, as you said, there's a change coming to that.

File: command/assets/connect.nomad (hunk 3 above)
Comment (tgross) [reply command/assets/connect.nomad]: Oh, because the `kind = "connect:<service>"` already includes the service name to pair them up?

Comment (nickethier) [reply command/assets/connect.nomad]: Yup!

File: command/assets/connect.nomad (hunk 2 above)
Comment (nickethier) [reply command/assets/connect.nomad]: No I think we can continue. Its covered it in the comment about referring to the task in `sidecar_service`

File: command/assets/connect.nomad (hunk 3 above)
Comment (tgross) [reply command/assets/connect.nomad]: Done.

**Comment by tgross** [comment]
@nickethier I think this is ready for re-review.

**Pull request merged**